src/zenml/cli/hub.py

```diff
@@ -0,0 +1,1203 @@
+    for tag in tags:
+        if tag.startswith(ZENML_HUB_INTERNAL_TAG_PREFIX):
+            logger.warning(
+                f"Tag '{tag}' will be ignored because it starts with "
+                f"disallowed prefix '{ZENML_HUB_INTERNAL_TAG_PREFIX}'."
+            )
+    return tags
+
+
+def _ask_for_tags() -> List[str]:
+    """Repeatedly ask the user for tags to assign to the plugin.
+
+    Returns:
+        A list of tags.
+    """
+    tags: List[str] = []
+    while True:
+        tag = click.prompt(
+            "(Optional) enter tags you want to assign to the plugin.",
+            default="",
+        )
+        if not tag:
+            return tags
+        if tag.startswith(ZENML_HUB_INTERNAL_TAG_PREFIX):
+            logger.warning(
+                "User-defined tags may not start with "
+                f"'{ZENML_HUB_INTERNAL_TAG_PREFIX}'."
+            )
+        else:
+            tags.append(tag)
+
+
+@hub.command("logs")
+@click.argument("plugin_name", type=str, required=True)
+@click.option(
+    "--version",
+    "-v",
+    type=str,
+    help=(
+        "Version of the plugin to get the logs for. If not provided, the "
+        "latest version will be used."
+    ),
+)
+def get_logs(
+    plugin_name: str,
+    version: Optional[str] = None,
+) -> None:
+    """Get the build logs of a plugin.
+
+    Args:
+        plugin_name: Name of the plugin.
+        version: Version of the plugin. If not provided, the latest version
+            will be used.
+    """
+    with event_handler(
+        event=AnalyticsEvent.ZENML_HUB_PLUGIN_LOGS,
+        metadata={
+            "plugin_name": plugin_name,
+            "plugin_version": version,
+        },
+    ) as analytics_handler:
+        client = HubClient()
+        analytics_handler.metadata["hub_url"] = client.url
+        plugin_name, author = _parse_plugin_name(plugin_name)
+        analytics_handler.metadata["plugin_name"] = plugin_name
+        analytics_handler.metadata["plugin_author"] = author
+        display_name = _plugin_display_name(plugin_name, version, author)
+
+        # Get the plugin from the hub
+        plugin = client.get_plugin(
+            plugin_name=plugin_name, plugin_version=version, author=author
+        )
+        if not plugin:
+            error(f"Could not find plugin '{display_name}' on the hub.")
+        analytics_handler.metadata["plugin_version"] = plugin.version
+
+        if plugin.status == PluginStatus.PENDING:
+            error(
+                f"Plugin '{display_name}' is still being built. Please try "
+                "again later."
+            )
+
+        if not plugin.build_logs:
+            logger.info(
+                f"Plugin '{display_name}' finished building, but no logs "
+                "were found."
+            )
+            return
+
+        for line in plugin.build_logs.splitlines():
+            if line.startswith("DEBUG"):
+                pass
```

Comment:
Why is this not logged as a debug log here? Shouldn't we allow users to set the verbosity of what they want to see in these logs?

Reply:
I replaced these with cli utils declare / warning / error already. Actually, we won't need the "DEBUG" and "WARNING" checks right now since we currently only log info/error there